### PR TITLE
Improved testing and query engine functionality

### PR DESCRIPTION
**Pull request by jsternberg**

### CODE REVIEW

File: influxql/point_test.go [hunk 1]
```diff
@@ -6,8 +6,155 @@ import (
...
+	}
+	if p.Value != c.Value {
+		t.Errorf("mismatched value: %v != %v", p.Value, c.Value)
+	}
+	if &p.Aux[0] == &c.Aux[0] {
+		t.Errorf("aux values share the same address: %v == %v", p.Aux, c.Aux)
+	} else if !deep.Equal(p.Aux, c.Aux) {
+		t.Errorf("mismatched aux fields: %v != %v", p.Aux, c.Aux)
+	}
+}
```
Comment (benbjohnson) [inline influxql/point_test.go:33]: Why not do a single `reflect.DeepEqual()` check and then `spew` the whole point if it doesn't match?

**Comment by benbjohnson** [comment]
Overall lgtm, one minor comment.

File: influxql/point_test.go (hunk 1 above)
Comment (jsternberg) [inline influxql/point_test.go:33]: No reason. I saw other code use `!deep.Equal()` in this package.

Comment (benbjohnson) [inline influxql/point_test.go:33]: Yeah, `deep.Equal()` works too. I had to add that because `reflect.DeepEqual()` considers `NaN == NaN` to be false.

**Pull request merged**